eslint/babel-eslint-parser/src/client.cjs

```diff
@@ -1,67 +1,105 @@
 const path = require("path");
 
-let send;
-
-exports.getVersion = sendCached("GET_VERSION");
-
-exports.getTypesInfo = sendCached("GET_TYPES_INFO");
-
-exports.getVisitorKeys = sendCached("GET_VISITOR_KEYS");
-
-exports.getTokLabels = sendCached("GET_TOKEN_LABELS");
-
-exports.maybeParse = (code, options) => send("MAYBE_PARSE", { code, options });
-
-function sendCached(action) {
-  let cache = null;
-
-  return () => {
-    if (!cache) cache = send(action, undefined);
-    return cache;
-  };
+const ACTIONS = {
+  GET_VERSION: "GET_VERSION",
+  GET_TYPES_INFO: "GET_TYPES_INFO",
+  GET_VISITOR_KEYS: "GET_VISITOR_KEYS",
+  GET_TOKEN_LABELS: "GET_TOKEN_LABELS",
+  MAYBE_PARSE: "MAYBE_PARSE",
+  MAYBE_PARSE_SYNC: "MAYBE_PARSE_SYNC",
+};
+
+class Client {
```

Comment:
I reorganized this file so that it can both run Babel using `worker_threads` or in the same thread. The `Client` class is the "public interface", while the `WorkerClient` and `LocalClient` provide the communication implementation.